src/components/modals/RateDialogsModal.tsx

```diff
@@ -0,0 +1,385 @@
+            if (!logDialog) {
+                history2 = []
+                missingLog = true
+            }
+            else {
+                const trainDialog = CLM.ModelUtils.ToTrainDialog(logDialog, this.props.actions, this.props.entities)
+                const teachWithHistory = await ((this.props.fetchHistoryThunkAsync(this.props.app.appId, trainDialog, this.props.user.name, this.props.user.id) as any) as Promise<CLM.TeachWithHistory>)
+                history2 = teachWithHistory.history
+            }
+        }
+        
+        // Find turn with first inconsistency
+        const maxLength = Math.max(history1.length, history2.length)
+        let stopTurn = -1
+        const replayError = new CLM.ReplayErrorTranscriptValidation()
+        for (let i = 0; i < maxLength; i = i + 1) {
+            const activity1 = history1[i] as BB.Activity
+            const activity2 = history2[i] as BB.Activity
+            if (!TranscriptUtils.isSameActivity(activity1, activity2)) {
+                if (activity1) {
+                    activity1.channelData.clData = {...activity1.channelData.clData, replayError  }
+                }
+                if (activity2) {
+                    activity2.channelData.clData = {...activity2.channelData.clData, replayError  }
+                }
+                stopTurn = i + 1
+                break
+            }
+        }
+
+        // Cut off history at first inconsistency
+        history1 = history1.slice(0, stopTurn)
+        history2 = history2.slice(0, stopTurn)
+
+        // Focuse same button (otherwise last choise will be active)
+        this.focusSameButton()
+
+        this.setState({
+            history1, 
+            history2,
+            missingLog,
+            webchatKey: this.state.webchatKey + 1,
+            scrollPosition: 0,
+            isFlipped: Math.random() >= 0.5
+        })
+    }
+
+    // Keep scroll position of two webchats in lockstep
+    @OF.autobind
+    onScrollChange(scrollPosition: number) {
+        this.setState({scrollPosition})
+    }
+
+    @OF.autobind
+    onSelectActivity(history: BotChat.Activity[] | undefined, activity: Activity) {
+        if (!history || history.length === 0) {
+            return
+        }
+
+        // Look up index
+        const selectedActivityIndex = history.findIndex(a => a.id === activity.id)
+        if (selectedActivityIndex > -1) {
+            this.setState({
+                selectedActivityIndex
+            })
+        }
+    }
+
+    render() {
+        const leftHistory = this.state.isFlipped ? this.state.history2 : this.state.history1
+        const rightHistory = this.state.isFlipped ? this.state.history1 : this.state.history2
+
+        return (
+            <div>
+                <OF.Modal
+                    isOpen={true}
+                    isBlocking={true}
+                    containerClassName="cl-modal cl-modal--compare-dialogs cl-modal--teach"
```

Comment:
This is not a teach modal. I would make another class that copies if you need those attributes. `cl-modal--rating` or something.  Would be cool to use this for merges.

Reply:
fixed